conans/test/command/test_package_test.py

```diff
@@ -0,0 +1,70 @@
+import unittest
+from conans.paths import CONANFILE
+from conans.test.utils.tools import TestClient
+
+conanfile = '''
+from conans import ConanFile
+
+class ConanLib(ConanFile):
+    name = "Hello"
+    version = "0.1"
+'''
+
+
+class TestPackageTest(unittest.TestCase):
+
+    def basic_test(self):
+        test_conanfile = '''
+from conans import ConanFile
+
+class TestConanLib(ConanFile):
+    def test(self):
+        pass
+'''
+        client = TestClient()
+        client.save({CONANFILE: conanfile,
+                     "test_package/conanfile.py": test_conanfile})
+        client.run("test_package -a lasote -c stable")
+        self.assertIn("Hello/0.1@lasote/stable: Configuring sources", client.user_io.out)
+        self.assertIn("Hello/0.1@lasote/stable: Generated conaninfo.txt", client.user_io.out)
+
+    def wrong_version_test(self):
+        test_conanfile = '''
+from conans import ConanFile
+
+class TestConanLib(ConanFile):
+    requires = "Hello/0.2@user/channel"
+    def test(self):
+        pass
+'''
+        client = TestClient()
+        client.save({CONANFILE: conanfile,
+                     "test_package/conanfile.py": test_conanfile})
+        client.run("test_package")
+        self.assertIn("Hello/0.1@user/channel: Configuring sources", client.user_io.out)
+        self.assertIn("Hello/0.1@user/channel: Generated conaninfo.txt", client.user_io.out)
```

Comment:
Yes, but the opposite is even more confusing. If you run this with 0.2 and have 0.1 in the root conanfile, the behavior is:
- Try to get 0.2 in the local cache, if it is there, use it!
- If not, try to get 0.2 from remotes, in order, when found, download and use it.
At the end, you are testing another package recipe different than the one you have!